src/media/headset.ts

```diff
@@ -0,0 +1,71 @@
+import { Observable } from 'rxjs';
+import { SdkMedia } from "..";
+import HeadsetService, { ConsumedHeadsetEvents, VendorImplementation} from 'softphone-vendor-headsets';
+
+export class SdkHeadset {
+    private media: SdkMedia;
+    private headsetLibrary: HeadsetService;
+    headsetEvents: Observable<ConsumedHeadsetEvents>;
+
+    constructor(media) {
+        this.media = media;
+        this.headsetLibrary = HeadsetService.getInstance({ logger: console });
+        this.headsetEvents = this.headsetLibrary.headsetEvents$;
+    }
+
+    getAudioDevice(newMicId: string): void {
+        const completeDeviceInfo = this.media.findCachedDeviceByIdAndKind(newMicId, 'audioinput' as MediaDeviceKind);
+        this.headsetLibrary.activeMicChange(completeDeviceInfo.label.toLowerCase());
+    }
+
+    getCurrentSelectedImplementation(): VendorImplementation {
+        return this.headsetLibrary.selectedImplementation;
+    }
+
+    showRetry(): boolean {
+        const selectedImplementation = this.getCurrentSelectedImplementation();
+        if (selectedImplementation.disableRetry) {
+            return false;
+        }
+
+        return selectedImplementation
+            && !selectedImplementation.isConnected
+            && ! selectedImplementation.isConnecting;
+    }
+
+    retryConnection(micLabel): void {
+        const selectedImplementation = this.getCurrentSelectedImplementation();
+        selectedImplementation.connect(micLabel.toLowerCase());
+    }
+
+    incomingCallRing(callInfo: { conversationId: string, contactName: string }, hasOtherActiveCalls) {
+        this.headsetLibrary.incomingCall(callInfo, hasOtherActiveCalls);
+    }
+
+    outgoingCall(callInfo: { conversationId: string, contactName: string }) {
+        this.headsetLibrary.outgoingCall(callInfo);
+    }
+
+    endCurrentCall(currentCallId: string) {
+        if (currentCallId) {
+            this.headsetLibrary.endCall(currentCallId);
+        }
+    }
+
+    endAllCalls() {
+        this.headsetLibrary.endAllCalls();
+    }
+
+    answerIncomingCall(currentCallId: string) {
```

Comment:
is currentCallId the same as conversationId? if so, let's stick with conversationId so there's no ambiguity.